db/migrate/20200313124507_add_type_to_reference.rb

```diff
@@ -0,0 +1,5 @@
+class AddTypeToReference < ActiveRecord::Migration[6.0]
+  def change
+    add_column :references, :type, :string
```

Comment:
FYI `type` is likely to conflict with Rails' Single Table Inheritance method. 

https://stackoverflow.com/questions/7134559/rails-use-type-column-without-sti